service/history/stateBuilder.go

```diff
@@ -129,6 +129,10 @@ func (b *stateBuilderImpl) applyEvents(domainID, requestID string, execution sha
 		switch event.GetEventType() {
 		case shared.EventTypeWorkflowExecutionStarted:
 			attributes := event.WorkflowExecutionStartedEventAttributes
+			domainEntry, err := b.domainCache.GetDomainByID(domainID)
+			if err != nil {
+				return nil, nil, nil, err
```

Comment:
this method signature definitely needs some revisiting. It takes about 6 params and returns 4 results

Reply:
agreed, let us do it one thing at a time :)